src/decoration.js

```diff
@@ -14,7 +14,11 @@ class WidgetType {
 
   map(mapping, span, offset, oldOffset) {
     let {pos, deleted} = mapping.mapResult(span.from + oldOffset, this.side < 0 ? -1 : 1)
-    return deleted ? null : new Decoration(pos - offset, pos - offset, this)
+    if (deleted) {
+      this.destroy()
+      return null
+    }
```

Comment:
is this the right moment to call the `destroy`? The DOM element wasn't removed yet.

Reply:
Ah good call. I realized this should be invoked by implementing `destroy` in `WidgetViewDesc`

Updated.